lmfit/minimizer.py

```diff
@@ -1457,19 +1508,20 @@ def least_squares(self, params=None, **kws):
             ret = least_squares(self.__residual, start_vals,
                                 bounds=(lower_bounds, upper_bounds),
                                 kwargs=dict(apply_bounds_transformation=False),
-                                **kws)
+                                max_nfev=2*self.max_nfev, **kws)
             result.residual = ret.fun
         except AbortFitException:
-            pass
+            ret = None
 
         # note: upstream least_squares is actually returning
         # "last evaluation", not "best result", but we do this
         # here for consistency, and assuming it will be fixed.
-        if not result.aborted:
-            result.residual = self.__residual(ret.x, False)
+        # if not result.aborted:
```

Comment:
unneeded leftover?

Reply:
Yeah, good point.  I'll tidy that up.

Reply:
can you also comment on the earlier review comments I left; it would be helpful to know your idea behind it when I look more closely at the code. Thanks!